typescript/infra/scripts/radix-helpers/generate-igp-manifest.ts

```diff
@@ -0,0 +1,147 @@
+import { Argv } from 'yargs';
+
+import rawGasPrices from '../../config/environments/mainnet3/gasPrices.json' with { type: 'json' };
+import rawTokenPrices from '../../config/environments/mainnet3/tokenPrices.json' with { type: 'json' };
+import { getArgs, withChains } from '../agent-utils.js';
+import { getEnvironmentConfig } from '../core-utils.js';
+
+export function withOriginChain<T>(args: Argv<T>) {
+  return args
+    .describe('originChain', 'The chain for which IGP prices are calculated')
+    .alias('o', 'origin-chain')
+    .default('originChain', '')
+    .demandOption('originChain');
+}
+
+const RADIX_ADDRESSES = {
+  igp: 'component_rdx1cznxpn5m3kutzr6jrhgnvv0x7uhcs0rf8fl2w59hkclm6m7axzlqgu',
+  // this is currently the deployer key and can be found in the secrets manager
+  igp_owner:
+    'account_rdx12x7j5plks60r73dus7znm6a4ryg8v89mw5dfkql74zvl0g3w60s822',
+  // In Radix the ownership-mechanism is controlled by a batch. Who-ever holds
+  // this badge is authorized to update/claim the IGP.
+  igp_owner_badge:
+    'resource_rdx1tkxgxg58vghrawkn4wvwz0h2vwqvhmy8e24rcn030re83schgt483g',
+};
+
+/**
+ * Generates the Radix manifest to update IGP configs.
+ */
+async function main() {
+  let { environment, chains, originChain } = await withOriginChain(
+    withChains(getArgs()),
+  ).argv;
+  const config = getEnvironmentConfig(environment);
+  const multiProvider = await config.getMultiProvider();
+
+  if (chains == undefined) {
+    chains = config.supportedChainNames;
+    console.log('No chains provided, generating config for all chains');
+  }
+
+  const gasPrices = rawGasPrices as unknown as {
+    [key: string]: { amount: string; decimals: number };
+  };
+  const tokenPrices = rawTokenPrices as unknown as { [key: string]: string };
+
+  const registry = await config.getRegistry(false);
+
+  const tokenPriceConfigs = chains.map((chain) => {
+    if (!tokenPrices[chain]) {
+      throw Error(`No token price found for ${chain}`);
+    }
+    if (!gasPrices[chain]) {
+      throw Error(`No gas price found for ${chain}`);
+    }
+
+    return {
+      name: chain,
+      domain_id: multiProvider.getDomainId(chain),
+      token_price: tokenPrices[chain],
+      gas_price_amount: gasPrices[chain].amount,
+      gas_price_decimals: gasPrices[chain].decimals,
+    };
+  });
+
+  if (!tokenPrices[originChain]) {
+    throw Error(`No token price found for ${originChain}`);
+  }
+
+  const tupleLines = [];
+  for (const entry of tokenPriceConfigs) {
```

Comment:
why not support the radix protocol type in igp.ts/gas-oracle.ts and then reuse the mapping we get with that? that way we don't have to configure this specially for radix? a la https://github.com/hyperlane-xyz/hyperlane-monorepo/blob/main/typescript/infra/scripts/sealevel-helpers/print-gas-oracles.ts